src/include/kdberrors.h

```diff
@@ -0,0 +1,313 @@
+/**
+ * @file
+ *
+ * @brief
+ *
+ * @copyright BSD License (see doc/LICENSE.md or http://www.libelektra.org)
+ */
+
+#ifndef KDBERRORS_H
+#define KDBERRORS_H
+
+#include <kdb.h>
+#include <kdbhelper.h>
+#include <kdblogger.h>
+#include <kdbmacros.h>
+
+#ifdef __cplusplus
+extern "C" {
+using Key = ckdb::Key;
+using KeySet = ckdb::KeySet;
+#endif
+
+#ifndef ELEKTRA_MODULE_NAME
+#define ELEKTRA_MODULE_NAME kdb
```

Comment:
When I implemented the new error codes I was not able to implement this for the core of elektra. Every plugin sets it correctly (guaranteed via the cmake LibAddPlugin). I remember also a longer discussion with @kodebach why this was difficult/ not possible but cannot remember the actual reasons.

Reply:
If you cannot remember, please remove and find out.

Reply:
The module name should be set in CMake (as a compiler argument), whenever we want something other than "kdb". AFAIK this is currently done in `add_plugin` and maybe also `add_lib`.

Reply:
Yes, but the question was if we need this default "kdb". Which module does not have any name set? We should fix that to not get the possibly wrong module name "kdb".

Reply:
I think `elektra-core` and `elektra-kdb` don't set a module name. In any case we should either have a default set or replace the `#define` with an error, otherwise the module name will be set to `ELEKTRA_MODULE_NAME` since the macro is simply not replaced.

Reply:
Yes, good objection. Of course it should be an error and not simply removed.